sql/mysqld.cc

```diff
@@ -9598,6 +9599,7 @@ static void set_server_version(void)
 #endif
   if (opt_log || opt_slow_log || opt_bin_log)
     strmov(end, "-log");                        // This may slow down system
+  strcpy(server_version_suffix, server_version + strlen(MYSQL_SERVER_VERSION));
```

Comment:
Check for overflow

Reply:
`server_version` has `SERVER_VERSION_LENGTH` bytes (and it's not checked for overflow) 
`server_version_suffix` is just a substring of `server_version` so if `server_version` is correct then `server_version_suffix` will not be overflowed
Should I change it anyway?